### DISCUSSION TITLE
Generating Code from Extended Introspection Types

### DISCUSSION THREAD

**Discussion by albanfellows** [dotansimha/graphql-code-generator#5503] [Q&A] [answered]
I am trying to use a feature of Hot Chocolate GraphQL server that allows the introspection types to be extended, but the code-generator only outputs the graphql default structure for introspection types.
We are generating typescript output for ReactJS and Apollo.

Our requirement is to have friendly and localised names for Enum values.
The extended type looks a bit like this:
```
type __EnumValue {
  name: String!
  description: String
  isDeprecated: Boolean!
  deprecationReason: String
  displayName: String!
  localisedDescription: String
  group: String
}
```
The generated code however does not include any of the added fields (`displayName`, `localisedDescription` or `group`)

Running the introspection query returns all the fields
e.g.
```
query {
    __type(name:"__EnumValue") {
      name
      fields {
        name
      }
    }
}
```
```
{
  "data": {
    "__type": {
      "name": "__EnumValue",
      "fields": [
        {
          "name": "name"
        },
        {
          "name": "description"
        },
        {
          "name": "isDeprecated"
        },
        {
          "name": "deprecationReason"
        },
        {
          "name": "localisedDescription"
        },
        {
          "name": "displayName"
        },
        {
          "name": "group"
        }
      ]
    }
  }
}
```
Is this something where the behaviour can be modified with configuration or a plugin, or is the handling of introspection types hard-coded to the ones specified in `graphql` and would require a code change to support?

**Comment by dotansimha** [comment, marked as answer]
I'm not sure we can support these, mainly because they are out-of-spec and fairly custom.

It really depends on how each extended type is implemented and what's the effect on the schema / generated types.

We are using the introspection only with `buildClientSchema` and then use that schema or AST for codegen. So introspection from our point of view is really temporary. So as long as the GraphQL spec and graphql-js doesn't support these - we can't support it either.
